src/main/java/com/aws/greengrass/certificatemanager/certificate/CISShadowMonitor.java

```diff
@@ -87,38 +108,48 @@ public void onConnectionResumed(boolean sessionPresent) {
     };
 
     /**
-     * Constructor.
+     * Construct a new CISShadowMonitor.
      *
      * @param mqttClient               IoT MQTT client
+     * @param ses                      Scheduled executor service
      * @param executorService          Executor service
      * @param deviceConfiguration      Device configuration
      * @param connectivityInfoProvider Connectivity Info Provider
      */
     @Inject
-    public CISShadowMonitor(MqttClient mqttClient, ExecutorService executorService,
+    public CISShadowMonitor(MqttClient mqttClient,
+                            ScheduledExecutorService ses,
+                            ExecutorService executorService,
                             DeviceConfiguration deviceConfiguration,
                             ConnectivityInfoProvider connectivityInfoProvider) {
-        this(mqttClient, null, null, executorService,
+        this(mqttClient, null, null, ses, executorService,
                 Coerce.toString(deviceConfiguration.getThingName()) + CIS_SHADOW_SUFFIX, connectivityInfoProvider);
         this.connection = new WrapperMqttClientConnection(mqttClient);
         this.iotShadowClient = new IotShadowClient(this.connection);
     }
 
-    CISShadowMonitor(MqttClient mqttClient, MqttClientConnection connection, IotShadowClient iotShadowClient,
-                     ExecutorService executorService, String shadowName,
+    CISShadowMonitor(MqttClient mqttClient,
+                     MqttClientConnection connection,
+                     IotShadowClient iotShadowClient,
+                     ScheduledExecutorService ses,
+                     ExecutorService executorService,
+                     String shadowName,
                      ConnectivityInfoProvider connectivityInfoProvider) {
         mqttClient.addToCallbackEvents(callbacks);
         this.connection = connection;
         this.iotShadowClient = iotShadowClient;
         this.executorService = executorService;
         this.shadowName = shadowName;
         this.connectivityInfoProvider = connectivityInfoProvider;
+        this.shadowProcessor = new ShadowProcessor(ses);
```

Comment:
this can, and should be injected

Reply:
does that work for inner classes?

Reply:
Yes, certainly

Reply:
sorry meant to say non-static inner class

Reply:
Yes should be fine.

Reply:
Haven't been able to get this to work, it looks like a chicken-and-egg issue, since nested non-static class depends on its parent


infinite loop resolving dependencies:
```
	at com.aws.greengrass.dependency.Context$Value.getOrCreateArgInstances(Context.java:520)
	at com.aws.greengrass.dependency.Context$Value.constructObjectWithInjection(Context.java:486)
	at com.aws.greengrass.dependency.Context$Value.get(Context.java:436)
	at com.aws.greengrass.dependency.Context.get(Context.java:100)
	at com.aws.greengrass.dependency.Context$Value.getOrCreateArgInstances(Context.java:520)
	at com.aws.greengrass.dependency.Context$Value.constructObjectWithInjection(Context.java:486)
	at com.aws.greengrass.dependency.Context$Value.get(Context.java:436)
	at com.aws.greengrass.dependency.Context.get(Context.java:100)
```

Reply:
So make it static or make it not an inner class?

Reply:
yeah, but I'm not sure if it's worth it. making it static requires passing in several fields from `CISShadowMonitor` that can't be injected (without adding extra wrappers i guess): `CISShadowMonitor#monitoredCertificateGenerators`, `CISShadowMonitor#shadowName`, `CISShadowMonitor#iotShadowClient`, inner class felt like the cleanest way to accomplish this. While I could just get rid of `ShadowProcessor` alltogether, it felt like a good way to group together functionality without making `CISShadowMontior` feel cluttered 🤷